docs/en/install-upgrade/installing-stack-demo-secure.asciidoc

```diff
@@ -0,0 +1,894 @@
++
+Just to avoid any possible name collisions, rename the existing `http.p12` file to distinguish it from the newly-created keystore:
++
+["source","shell"]
+----
+mv http.p12 http-old.p12
+----
+
+. Unzip the generated `elasticsearch-ssl-http.zip` archive:
++
+["source","shell"]
+----
+unzip elasticsearch-ssl-http.zip
+----
+
+. When the archive is unpacked, the certificate files are located in separate directories for each {es} node and for the {kib} node.
++
+You can run a recursive `ls` command to view the file structure:
++
+["source","shell"]
+----
+ls -lR elasticsearch kibana
+----
++
+["source","shell"]
+----
+elasticsearch:
+total 0
+drwxr-xr-x. 2 root root 56 Dec 12 19:13 mynode-es1
+drwxr-xr-x. 2 root root 72 Dec 12 19:04 mynode-es2
+drwxr-xr-x. 2 root root 72 Dec 12 19:04 mynode-es3
+
+elasticsearch/mynode-es1:
+total 8
+-rw-r--r--. 1 root root 1365 Dec 12 19:04 README.txt
+-rw-r--r--. 1 root root  845 Dec 12 19:04 sample-elasticsearch.yml
+
+elasticsearch/mynode-es2:
+total 12
+-rw-r--r--. 1 root root 3652 Dec 12 19:04 http.p12
+-rw-r--r--. 1 root root 1365 Dec 12 19:04 README.txt
+-rw-r--r--. 1 root root  845 Dec 12 19:04 sample-elasticsearch.yml
+
+elasticsearch/mynode-es3:
+total 12
+-rw-r--r--. 1 root root 3652 Dec 12 19:04 http.p12
+-rw-r--r--. 1 root root 1365 Dec 12 19:04 README.txt
+-rw-r--r--. 1 root root  845 Dec 12 19:04 sample-elasticsearch.yml
+
+kibana:
+total 12
+-rw-r--r--. 1 root root 1200 Dec 12 19:04 elasticsearch-ca.pem
+-rw-r--r--. 1 root root 1306 Dec 12 19:04 README.txt
+-rw-r--r--. 1 root root 1052 Dec 12 19:04 sample-kibana.yml
+----
+
+. In the directory where you unzipped the archive, replace your existing keystore with the new keystore. The location of your certificate directory may be different than what is shown here, depending on the installation method you chose.
++
+Run the `mv` command, replacing `<es1-hostname>` with the hostname of your initial {es} node:
++
+["source","shell"]
+----
+mv elasticsearch/<es1-hostname>/http.p12 /etc/elasticsearch/certs/
+----
+
+. Because this is a new keystore, the {es} configuration file needs to be updated with the path to its location. Open `/etc/elasticsearch/elasticsearch.yml` and update the HTTP SSL settings with the new path:
++
+["source","yaml"]
+----
+xpack.security.http.ssl:
+  enabled: true
+  #keystore.path: certs/http.p12
+  keystore.path: /etc/elasticsearch/certs/http.p12
+----
+
+. Since you also generated a new keystore password, the {es} keystore needs to be updated as well. From the {es} installation directory, first remove the existing HTTP keystore entry:
++
+["source","shell"]
+----
+sudo usr/share/elasticsearch/bin/elasticsearch-keystore remove xpack.security.http.ssl.keystore.secure_password
+----
+
+. From the {es} installation directory, add the updated HTTP keystore password, using the password you generated for this keystore:
```

Comment:
Can just start this paragraph with "Add the updated HTTP...". The first part isn't necessary.